### PR TITLE
Try and log the VM info + console log on failure

### PR DESCRIPTION
**Pull request by hawkowl**
Fixes: https://issues.redhat.com/browse/ARO-5368

Dumps the VM info + console logs on failure so that we don't need to run the Geneva Action or have the control plane still around to get it.

### CODE REVIEW

**Review by jaitaiwan — DISMISSED** [review]
LGTM

**Review by mociarain — APPROVED** [review]
2 small things. LGTM

File: pkg/util/steps/runner.go [hunk 1]
```diff
@@ -21,6 +21,11 @@ func FriendlyName(f interface{}) string {
 	return runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
 }
 
+// FriendlyName returns a "friendly" stringified name of the given func.
+func ShortFriendlyName(f interface{}) string {
```
[the remark this answers is not in the document]
Comment (jaitaiwan) [reply pkg/util/steps/runner.go:25]: Addressed in https://github.com/Azure/ARO-RP/pull/3422

[the remark this answers is not in the document]
Comment (mociarain) [reply pkg/util/steps/runner.go:25]: For reference this is the relevant thread: https://github.com/Azure/ARO-RP/pull/3422#discussion_r1508195616

**Comment by hawkowl** [comment]
See new PR #3629

**Pull request closed without merging**